src/game_main.cpp

```diff
@@ -1057,6 +1062,9 @@ int GameMain(const CmdLineSetup_t &setup)
             if(LevelRestartRequested && Checkpoint.empty())
                 StartWarp = lastWarpEntered; // When restarting a level (after death), don't restore an entered warp on checkpoints
 
+            if(IsHubLevel && StartWarp > 0) // Save the warp where player entered the hub
+                ReturnWarpSaved = StartWarp;
+
```

Comment:
```
else if(IsHubLevel && ReturnWarp == 0)
    ReturnWarpSaved = 0;
```

Reply:
Good note!

Reply:
Actually, a further note: at this point in the code, should `ReturnWarp` refer to the sub-hub, or the main hub?